validator/sawtooth_validator/execution/executor.py

```diff
@@ -292,33 +300,47 @@ def _execute_schedule(self):
                     is_valid=False,
                     context_id=None)
                 continue
-            content = processor_pb2.TpProcessRequest(
+
+            process_request = processor_pb2.TpProcessRequest(
                 header=header,
                 payload=txn.payload,
                 signature=txn.header_signature,
-                context_id=context_id).SerializeToString()
+                context_id=context_id,
+                header_bytes=txn.header)
 
             # Since we have already checked if the transaction should be failed
             # all other cases should either be executed or waited for.
             self._execute(
                 processor_type=processor_type,
-                content=content,
-                signature=txn.header_signature)
+                process_request=process_request)
 
         self._done = True
 
-    def _execute(self, processor_type, content, signature):
+    def _execute(self, processor_type, process_request):
         try:
             processor = self._processor_manager.get_next_of_type(
                 processor_type=processor_type)
         except WaitCancelledException:
             LOGGER.exception("Transaction %s cancelled while "
                              "waiting for available processor",
-                             content.signature)
+                             process_request.signature)
             return
 
+        if processor.request_header_style() == \
```

Comment:
We need some more comments here; currently it's hard to read this block because it's removing instead of setting. So maybe adding the context would help - "Previously, both header and header_bytes fields were filled into the protobuf object, because the earlier code does not have the correct context to determine which to fill in. So here, we make sure only the correct field contains data."

Reply:
Done